AxonDeepSeg/apply_model.py

```diff
@@ -1,73 +1,166 @@
 from pathlib import Path
+import os
+import numpy as np
+import torch
+from loguru import logger
+from typing import List, Literal, Dict, NoReturn
 
 # AxonDeepSeg imports
-import AxonDeepSeg.ads_utils as ads
 from AxonDeepSeg.visualization.merge_masks import merge_masks
-from config import axon_suffix, myelin_suffix, axonmyelin_suffix
-from loguru import logger
+from AxonDeepSeg import ads_utils
+from config import axon_suffix, myelin_suffix, axonmyelin_suffix, nnunet_suffix
+from AxonDeepSeg.params import intensity
 
-from ivadomed import inference as imed_inference
+os.environ['nnUNet_raw'] = 'UNDEFINED'
+os.environ['nnUNet_results'] = 'UNDEFINED'
+os.environ['nnUNet_preprocessed'] = 'UNDEFINED'
+from nnunetv2.inference.predict_from_raw_data import nnUNetPredictor
 
-default_overlap = 48
+def get_checkpoint_name(checkpoint_folder_path: Path) -> str:
+    '''
+    Get the name of the checkpoint file in the given folder, with priority for 
+    best validation checkpoint.
 
-def axon_segmentation(
-                    path_acquisitions_folders, 
-                    acquisitions_filenames,
-                    path_model_folder,
-                    acquired_resolution,
-                    overlap_value=None,
-                    no_patch=False,
-                    gpu_id=0,
-                    verbosity_level=0
-                    ):
+    Parameters
+    ----------
+    checkpoint_folder_path : pathlib.Path
+        Path to the folder containing the .pth checkpoint file.
+
+    Returns
+    -------
+    str
+        Name of the checkpoint file, e.g. 'checkpoint_best.pth'.
     '''
-    Segment images using IVADOMED.
-    :param path_acquisitions_folders: the directory containing the images to segment.
-    :param acquisitions_filenames: filenames of the images to segment.
-    :param path_model_folder: path to the folder of the IVADOMED-trained model.
-    :param overlap_value: the number of pixels to be used for overlap between patches when doing prediction.
-    Higher value means less border effects but more time to perform the segmentation.
-    :param acquired_resolution: isotropic pixel size of the acquired images.
-    :param no_patch: If True, the image is segmented without using patches. Default: False. This parameter supersedes
-    the "overlap_value" parameter. This option may not be suitable with large images depending on computer RAM capacity.
-    :param gpu_id: Number representing the GPU ID for segmentation if available. Default 0.
-    :param verbosity_level: Level of verbosity. The higher, the more information is given about the segmentation
-    process.
-    :return: Nothing.
+    if (checkpoint_folder_path / 'checkpoint_best.pth').exists():
+        return 'checkpoint_best.pth'
+    elif (checkpoint_folder_path / 'checkpoint_final.pth').exists():
+        return 'checkpoint_final.pth'
+    else:
+        # just return the first .pth file found
+        next((checkpoint_folder_path).glob('*.pth')).name
```

Comment:
self: line not covered by tests

Reply:
I found that this line had a bug while making tests for this function; it did not return a filename like the other two if/elif cases.

I fixed it in [bddf889](https://github.com/axondeepseg/axondeepseg/pull/800/commits/bddf889396aade6facfa964ec950c6d62c072bde), and also made so that it returns the checkpoint with the last modification time instead of the next arbitrarily sorted glob item. I'm hoping that download tests doesn't change the modification times, I'll wait until the CI tests pass before closing this comment

Reply:
Hum - looks like in Windows and Mac, the modification time is preserved, but not for Unix. I'll switch to simply take the last file after sorting by filename, assuming there's some an incrementing order in the checkpoint outputs filenames

Reply:
https://github.com/axondeepseg/axondeepseg/actions/runs/10371634002/job/28712391448?pr=800

Reply:
Done in [d479b6c](https://github.com/axondeepseg/axondeepseg/pull/800/commits/d479b6cdf7843d4394afb17bad6de2395d94a05e)